docs/proposals/odo-config-file.md

```diff
@@ -0,0 +1,132 @@
+# odo config file
+
+## Abstract
+When developers work on creating an application, there's 2 phases they go though. First is creating the application initial implementation from scratch, and then iterate development on an existing application. While the developer works on the application, he might take some decisions on how the application should behave that should be preserved over time, but that are not source code but specific to how the application should behave in relation to the platform where it's running or in relation to the tool used to develop it.
+
+IDEs often use a configuration file to hold specific configuration for an application, workspace or the whole tool. These files are stored under the same control version system as where the source code lives so that anytime a developer checks out that repository has control on how to configure that tool to work with that source code.
+
+## Motivation
+Provide a way to preserve conscisous decisions the developer has made that are orthogonal to the code but relevant to the tool/platform. Any user that continues working on an existing application don't need to know the decisions the original developer took, as these are preserved along with the source code.
+
+## Use Cases
+
+1. As a user, I want to be able to preserve some configuration I've done about my current component along with my source code
+1. As a user, I want to be able to use configuration stored with my source code to help me be more productive
+1. As a user, I want to be able to manage explicit configuration that is saved with my source code
+
+## Design overview
+We need to provide a local file, `.odo` (name subject to change) that will hold specific configuration for the source code to instruct how a component should be configured.
+
+### Save user provided explicit flags
+The user should not need to explicitly save the configuration in this file, but the odo CLI should keep the values explicitly used by a user into this file.
+
+- If the file does not exist, it should create it.
+- If the value is already present in the file, it should overwrite it.
+
+```
+odo create wildfly example --memory=1GB
+
+# This should set value in .odo config file   (memory=2GB)
```

Comment:
also, is this `# This should set value in .odo config file   (memory=2GB)` 1GB instead of 2GB?

Reply:
yes, that is typo ;-)